### ISSUE TITLE
Fix ACU2.0 Watchdog

### ISSUE THREAD

**Issue by CPollreis** [UMSAE-Formula-Electric/ACU2.0#17]
# Overview
The watchdog needs to be tested on the ACU because we did e-tech with the ACU watchdog disabled during ePBR24 comp. The VCU watchdog was enabled during comp so use that as a reference.

Tony's one note about this issue is that "the problem with the watchdogs on the ACU is probably the timers" so reference the timers on the VCU.

# Testing
Once you feel like the ACU watchdog is in a good spot (similar to the VCU watchdog), test it by doing the following (Tony's suggestions):

- have all tasks except 1 disabled with the watchdog enabled and wait till the watchdog resets itself or it doesn't reset itself
- once you go through each one individually then run the watchdog with all the tasks enabled and see if you're resetting
  - From there, adjust the watchdog so we don't unintentionally reset

**Comment by Mason-Pronger** [comment]
Once the test bench is in place I'll begin debugging.

**Comment by CPollreis** [comment]
- need to understand watchdog concept first, can come back to this task after that

**Comment by ceddd22** [comment]
I originally tried testing the ACU Watchdog through Renode but I kept running into errors so I moved to testing with my dev board. Further Renode work will need to be done on my end.

I didn't make any changes to the code and I successfully tested and validated the WDT implementation on my physical microcontroller through three primary test cases:
1. Global Timeout: To ensure that the watchdog timer is counting properly, I commented out the main WDT refresh call and confirmed that the microcontroller does reset as expected when the refresh doesn't happen. So even if all tasks are "kicking" the watchdog, the microcontroller still resets (I have an LED blinking when a reset occurs.)
2. Task-Level: I then manually disabled the "kick" functions for individual tasks one by one. I uncommented out the main WDT refresh call, and I confirmed that the WDT correctly identified the "broken or hung" tasks and triggered a system reset.
3. Normal Operation: I verified that the system works with all tasks active and the watchdog refreshing normally.

List of tasks used by watchdog:
#define CAN_TX_TASK_ENABLED 1 //wd
#define COOLING_TASK_ENABLED 1 //wd
#define VCU_HRTBEAT_TASK_ENABLED 1 //wd
#define ACU_STATE_TASK_ENABLED 1 //wd
#define CAN_RX_TASK_ENABLED 1 //wd
#define DEFAULT_TASK_ENABLED 1 //wd
#define LED_TASK_ENABLED 1 //wd
#define WATCH_DOG_TASK_ENABLED 1 //wd
#define BMS_CAN_COMM_TASK_ENABLED 1 //wd
#define VCU_CAN_COMMS_TASK_ENABLED 1 //wd

Results: Without making any changes to the current implementation, the watchdog works as intended.

**Comment by CPollreis** [comment]
Yeah your tests look thorough to me so I'll close this for now and we'll re-open this issue if it occurs again
